services/JSONFileService.js

```diff
@@ -0,0 +1,30 @@
+const fs = require("fs").promises;
+
+class JSONFileService {
+  constructor(filePath) {
+    this.filePath = filePath;
+  }
+
+  getData() {
+    return fs
+      .readFile(this.filePath, "utf8")
+      .then((data) => JSON.parse(data))
+      .catch((error) => {
+        throw error;
+      });
+  }
+
+  updateData(newData) {
+    const data = JSON.stringify(newData, null, 2);
```

Comment:
i think it is better constatly set new data into `{ tasks: newData }`. and read it from file same way - `JSON.parse(data).tasks`